src/cli-discover-silos.ts

```diff
@@ -0,0 +1,92 @@
+#!/usr/bin/env node
+
+import { mapSeries } from 'bluebird';
+import yargs from 'yargs-parser';
+import { logger } from './logger';
+import colors from 'colors';
+import { ADMIN_DASH } from './constants';
+import { GraphQLClient } from 'graphql-request';
+import { ADD_SILO_DISCOVERY_RESULTS } from './gqls';
+import { SILO_DISCOVERY_FUNCTIONS, SupportedPlugin } from './plugins';
+import { isSupportedPlugin } from './plugins/typeguards';
+import { SiloDiscoveryRawResults } from './plugins/types';
+
+const CHUNK_SIZE = 1000;
+
+/**
+ * Sync data silo configuration from Transcend down locally to disk
+ *
+ * Dev Usage:
+ * yarn ts-node ./src/cli-discover-silos.ts --scanPath=./myJavascriptProject \
+ *   --auth=asd123 \
+ *   --ignoreDirs=build_directories_to_ignore \
+ *   --pluginType="javascriptPackageJson" \
+ *   --pluginId=abcdefgh
+ *
+ * Standard usage
+ * yarn tr-scan  --scanPath=./myJavascriptProject --auth=asd123
+ */
+async function main(): Promise<void> {
+  // Parse command line arguments
+  const {
+    scanPath = '.',
+    ignoreDirs = '',
+    transcendUrl = 'https://api.transcend.io',
+    pluginType = '',
+    // TODO: https://transcend.height.app/T-14107 - Remove need to pass in pluginId for silo discovery cli
```

Comment:
see other comment, but i think we should do the opposite, provide pluginId and have that derive the `pluginType`, this would involve modifying a graphql query which is fine